### ISSUE TITLE
We should not require mandatory flag for meshConfig

### ISSUE THREAD

**Issue by linsun** [istio/istio#2905]
From @ldemailly :> What does not work out of the box and we must fix is that kube-inject now requires a new mandatory flag 

@linsun: I agree with you on that, which I complained to Jason earlier today too.  I think requiring the mandatory flag is bad as users will need to change their micro services deployment script now when moving from 0.4 or prior to 0.5.

@christian-posta: +1 not good user experience either

Test sample:
```
$ ./istioctl kube-inject -f ~/Downloads/istio-0.4.0/samples/bookinfo/kube/bookinfo.yaml
Error: --meshConfigFile or --meshConfigMapName must be set
```
Ideally, I would like either of these to be optional and read the default mesh config if neither is specified.

cc @ayj

**Comment by linsun** [comment]
https://github.com/istio/istio/pull/2898

**Comment by ayj** [comment]
Fixed by https://github.com/istio/istio/issues/2905

**Comment by linsun** [comment]
fixed by #2898
